app/src/server/api/routers/payments.router.ts

```diff
@@ -93,6 +94,20 @@ export const paymentsRouter = createTRPCRouter({
           defaultPaymentMethodId = availablePaymentMethodId;
         }
 
+        // Update Rate Limit after a payment method is added
+        if (paymentMethods.data.length) {
```

Comment:
It's a weird place to put this logic, but by adding a payment method, we call Stripe directly, not through our API. After that, we hit this endpoint to fetch payment methods and set a default one.